### DISCUSSION TITLE
Having difficulty with requestOptions

### DISCUSSION THREAD

**Discussion by NickStull** [wellyshen/use-places-autocomplete#828] [Q&A]
From what I am understanding you can use requestOptions as a way to set parameters for the api call. Reading the Places docs it seems that I can restrict the search to a specific type. I would like ONLY specific addresses to be returned, meaning no establishments, just cities, etc. This would be done using types=address.

This is what I have in my autocomplete component:

`} = usePlacesAutocomplete({
        requestOptions: {
          types: "address"
        },
        debounce: 300,
    });`

This creates very unexpected results, for example if I type the number 3 in the box I get a list of results. If I type the number 8 I get zero results at all. 

Is my code not correct here? or is there a bug somewhere?

Thank you a ton for the help.
@wellyshen

**Comment by NickStull** [comment]
I figured out a workaround, but this technically is still unanswered. There is something funky going on with the requestOptions. I tried a few different options from google docs and none of them seem to work, and there are not any examples of working options that I can find in the docs for the package, just "options go here". It very well could be my syntax is wrong, but I can't find anything to go on.

I basically built a workaround by then taking the geocode and rather than look for the lat/lon like in the example given, I grabbed the details, pulled out the address_components, and wrote a quick little algorithm to sort through and only allow the full addresses to used.

I'm leaving this up in the case that the owner of this repo or someone else comes by with an actual solution.

**Reply by wellyshen** [reply]
@NickStull Sorry for the late reply. According to the [doc](https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompletionRequest.types). The `types` receives an array of strings, so the corrected way of usage as follows:

```js
const result = usePlacesAutocomplete({ requestOptions: { types: ["address"] } });
```

**Reply by orbiteleven** [reply]
Heya @wellyshen,

I'm doing something similar. My setup is:
```typescript
const result = usePlacesAutocomplete({
  requestOptions: { types: ['country'] },
})
```

and the resulting url is `https://maps.googleapis.com/maps/api/place/js/AutocompletionService.GetPredictionsJson?1sden&4sen-US&9scountry&15e3&21m1&2e1&callback=_xdc_._99a7t3&key=<redacted>&token=<redacted>`

but I'm getting the following response: 
```jsonc
/**/_xdc_._99a7t3 && _xdc_._99a7t3( {
   "predictions" : [],
   "status" : "INVALID_REQUEST"
})
 ```

Awesome work, BTW :)

**Reply by wellyshen** [reply]
@orbiteleven The correct way should be [this](https://github.com/wellyshen/use-places-autocomplete/discussions/828).

**Reply by orbiteleven** [reply]
@wellyshen um... not sure if this is a recursion joke or just RTFM, but [this](https://github.com/wellyshen/use-places-autocomplete/discussions/828) is the discussion on which I'm asking for more help 😆

AFAIKT this:
```typescript
const result = usePlacesAutocomplete({ requestOptions: { types: ["address"] } });
```
is only different than this:
```typescript
const result = usePlacesAutocomplete({ requestOptions: { types: ['country'] } });
```
by way of the type being used. `country` is one of the available types from [the docs](https://developers.google.com/maps/documentation/places/web-service/supported_types#table3).

In looking at [this line](https://github.com/wellyshen/use-places-autocomplete/blob/master/src/usePlacesAutocomplete.ts#L132), it seems like it should work, though I'm still struggling.

***Edited to remove CodeSandbox that I've taken down so I don't thrash my Google API limits***

**Reply by orbiteleven** [reply]
...and now I realize my error: apparently one cannot select `country` directly, only the following: `geocode`, `address`, `establishment`, `(regions)` (which includes, among other types, `country`) or `(cities)`.

So it *was* RTFM, though in my defense I think the Google docs could be more explicit here. 

Anyway, thanks for the awesome library!

![](https://media.giphy.com/media/jUwpNzg9IcyrK/giphy.gif)

**Reply by wellyshen** [reply]
Yea... many ppl are confused by the doc, even me lol.

**Reply by avedisgmb** [reply]
I don't know if I get it right. With use places autocomplete we can't apply filter of types='address' & country:'us' at the same time ?

Because I want to filter only the places from USA that have specific address with this code

```
usePlacesAutocomplete({
 requestOptions: {
   types: ["address"],
   componentRestrictions: {
     country: "us",
   },
 }
```
 
But when typing `Holly` I'm getting some results that don't have specific address because they are cities. What I'm doing wrong in this case?  Does the hook enabled this ?

**Reply by ashahid1-wooliesx** [reply]
I am facing the same issue 
`componentRestrictions: {
     country: "au",
   },`

not working and showing predictions with countries all over the globe

**Reply by Zaimwa9** [reply]
did you fix @ab1303 ? Having the same issue i.e country not working

**Reply by ashahid1-wooliesx** [reply]
Hi @Zaimwa9 ,
Its been sometime I worked on that project
This is how I set that up and its working for me.

```
const {
    init: initialiseGoogleMap,
    suggestions: { data },
    setValue: addressAutoCompleteSetValue,
  } = usePlacesAutocomplete({
    requestOptions: {
      componentRestrictions: {
        country: 'AU',
      },
    },
    initOnMount: false,
  });

  React.useEffect(() => {
    if (window.google && window.google.maps) initialiseGoogleMap();
  }, [initialiseGoogleMap]);
```

**Reply by raja-muhammad-asher** [reply]
Hi @Zaimwa9, This works perfectly fine. All you need to do is to empty your session storage. Here is the description on [StackOverFlow](https://stackoverflow.com/questions/70672199/filtering-only-address-from-usa-with-use-places-autocomplete)

**Reply by paoloLigsay** [reply]
@raja-muhammad-asher Thanks, clearing cache definitely worked!
